nav2_system_tests/src/planning/test_planner_is_path_valid.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. Reserved.
+
+#include <gtest/gtest.h>
+#include <memory>
+#include <vector>
+
+#include "nav2_msgs/srv/is_path_valid.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "planner_tester.hpp"
+#include "nav2_util/lifecycle_utils.hpp"
+
+using nav2_system_tests::PlannerTester;
+using nav2_util::TestCostmap;
+
+TEST(testIsPathValid, testIsPathValid)
+{
+  auto planner_tester = std::make_shared<PlannerTester>();
+  planner_tester->activate();
+  // load open space cost map which is 10 by 10
+  planner_tester->loadSimpleCostmap(TestCostmap::top_left_obstacle);
+
+  nav_msgs::msg::Path path;
+
+  bool is_path_valid = planner_tester->checkPathValid(path);
+  EXPECT_FALSE(is_path_valid);
+
+  for (float i = 0; i < 10; i += 0.05) {
+    for (float j = 0; j < 10; j += 0.05) {
+      geometry_msgs::msg::PoseStamped pose;
+      pose.pose.position.x = i;
+      pose.pose.position.y = j;
+      path.poses.push_back(pose);
```

Comment:
Just a small thought. We have this nice function that creates the plan for you? https://github.com/ros-planning/navigation2/blob/609342fa27016a1677768f93fb3c44dec98b1f55/nav2_system_tests/src/planning/planner_tester.hpp#L77

Maybe it would be handy?

Reply:
The goal of the first section it to create a invalid path. The `createPath` will create a valid path. 
I think separating them is good. But, honestly either way would probably be fine.